### ISSUE TITLE
[REGRESSION]: Suddenly stopped working despite nothing having changed (`experimentalLoader.js:load` did not call the next hook in its chain and did not explicitly signal a short circuit)

### ISSUE THREAD

**Issue by NatoBoram** [microsoft/playwright#15717]
**Context:**
- GOOD Playwright Version: 1.23.2
- BAD Playwright Version: 1.23.2
- Operating System: Linux
- Extra: Fails on both GitHub actions and on PopOS

**Code Snippet**

Help us help you! Put down a short code snippet that illustrates your bug and
that we can run and debug locally. For example:

```typescript
import { expect, test } from '@playwright/test';

test('index page has expected header', async ({ page }) => {
	await page.goto('/');
	expect(await page.textContent('header'));
});
```

I mean, really, nothing changed.

Yesterday I ran `playwright test` on `main`, it worked, and today it doesn't, but there hasn't been any commit.

**Describe the bug**

```typescript
Error: "file:///node_modules/.pnpm/@playwright+test@1.23.2/node_modules/@playwright/test/lib/experimentalLoader.js 'load'" did not call the next hook in its chain and did not explicitly signal a short circuit. If this is intentional, include `shortCircuit: true` in the hook's return.
    at new NodeError (node:internal/errors:387:5)
    at ESMLoader.load (node:internal/modules/esm/loader:624:13)
    at ESMLoader.moduleProvider (node:internal/modules/esm/loader:472:11)
    at link (node:internal/modules/esm/module_job:67:21) {
  code: 'ERR_LOADER_CHAIN_INCOMPLETE'
}
 ELIFECYCLE  Test failed. See above for more details.
```

**Comment by yury-s** [comment]
The error is introduced by Node v18.6, it was fixed on the playwright end in https://github.com/microsoft/playwright/pull/15694 which is going to be released in 1.24 (we are not planning to patch 1.23 release with the fix as 1.24 is going to be released in the next couple weeks). Before that you have two options to mitigate the problem:
* roll back your node.js to a version before 18.6 and make sure it doesn't update under your feet;
* use @playwright/test@next which is built from the main branch of playwright and updated daily.

**Comment by yury-s** [comment]
This should be fixed in v1.23.4

**Comment by NatoBoram** [comment]
Thanks!
